src/main/java/com/mg105/use_cases/ChestInteractor.java

```diff
@@ -65,6 +65,7 @@ public TreasureChest verifyChest() {
 
     public void getChestItem() {
         TreasureChest chest = verifyChest();
+        gameState.getCurrentRoom().getChests().remove(chest);
```

Comment:
I think this should go last because at this point in the code the chest could be null

Reply:
It should be fine.